crates/spidermonkey-embedding-splicer/src/splice.rs

```diff
@@ -757,32 +757,50 @@ fn synthesize_import_functions(
             .get_fn_modifier(coreabi_get_import_fid)
             .unwrap();
 
-        // walk until we get to the const representing the table index
-        let mut table_instr_idx = 0;
-        for (idx, instr) in builder.body.instructions.iter_mut().enumerate() {
-            if let Operator::I32Const { value: ref mut v } = instr.op {
-                // we specifically need the const "around" 3393
-                // which is the coreabi_sample_i32 table offset
-                if *v < 1000 || *v > 5000 {
-                    continue;
+        // Find the I32Const base index and compute the delta to new base
+        let mut table_instr_idx = 0usize;
+        let mut delta: i32 = 0;
+        {
+            let ops_ro = builder.body.instructions.get_ops();
```

Comment:
I think you should be able to revert the changes to this loop and instead change `instructions.iter_mut()` to `instructions.get_ops().iter_mut()` in (the original) line 762.

Reply:
`builder.body.instructions.get_ops()` returns an immutable reference so you can't use `iter_mut()` on it

Reply:
Ah, right. There's a `get_ops_mut()` that should hopefully work, though